javalib/src/main/scala/java/lang/String.scala

```diff
@@ -1,42 +1,951 @@
+    }
+    if (thisStart < 0 || count - thisStart < length) {
+      return false
+    }
+    if (length <= 0) {
+      return true
+    }
+    val o1 = offset + thisStart
+    val o2 = string.offset + start
+    for (i <- 0 until length if value(o1 + i) != string.value(o2 + i)) {
+      return false
+    }
+    true
+  }
+
+  def regionMatches(ignoreCase: scala.Boolean,
+                    thisStart: Int,
+                    string: _String,
+                    start: Int,
+                    length: Int): scala.Boolean = {
+    if (!ignoreCase) {
+      return regionMatches(thisStart, string, start, length)
+    }
+    if (string != null) {
+      if (thisStart < 0 || length > count - thisStart) {
+        return false
+      }
+      if (start < 0 || length > string.count - start) {
+        return false
+      }
+      thisStart += offset
+      start += string.offset
+      val end = thisStart + length
+      var c1: Char = 0
+      var c2: Char = 0
+      val target = string.value
+      while (thisStart < end) {
+        if ((c1 = value(thisStart += 1)) != (c2 = target(start += 1)) &&
+          toUpperCase(c1) != toUpperCase(c2) &&
+          toLowerCase(c1) != toLowerCase(c2)) {
+          return false
+        }
+      }
+      return true
+    }
+    throw new NullPointerException()
+  }
+
+  def replace(oldChar: Char, newChar: Char): _String = {
+    var index = indexOf(oldChar, 0)
+    if (index == -1) {
+      return this
+    }
+    val buffer = Array.ofDim[Char](count)
+    System.arraycopy(value, offset, buffer, 0, count)
+    do {
+      buffer(index += 1) = newChar
+    } while ((index = indexOf(oldChar, index)) != -1);
+    new _String(0, count, buffer)
+  }
+
+  def replace(target: CharSequence, replacement: CharSequence): _String = {
+    if (target == null) {
+      throw new NullPointerException("target should not be null")
+    }
+    if (replacement == null) {
+      throw new NullPointerException("replacement should not be null")
+    }
+    val ts = target.toString
+    var index = indexOf(ts, 0)
+    if (index == -1) return this
+    val rs = replacement.toString
+    if ("" == ts) {
+      val buffer = new StringBuilder(count + (rs.length * (count + 1)))
+      buffer.append(rs)
+      for (i <- 0 until count) {
+        buffer.append(value(offset + i))
+        buffer.append(rs)
+      }
+      return buffer.toString
+    }
+    val buffer = new StringBuilder(count + rs.length)
+    val tl = target.length
+    var tail = 0
+    do {
+      buffer.append(value, offset + tail, index - tail)
+      buffer.append(rs)
+      tail = index + tl
+    } while ((index = indexOf(ts, tail)) != -1);
+    buffer.append(value, offset + tail, count - tail)
+    buffer.toString
+  }
+
+  def startsWith(prefix: _String): scala.Boolean = startsWith(prefix, 0)
+
+  def startsWith(prefix: _String, start: Int): scala.Boolean = {
+    regionMatches(start, prefix, 0, prefix.count)
+  }
+
+  def sub_String(start: Int): _String = {
```

Comment:
This should be `subString`. Names of the methods don't change, only the name of the `String` type.